bandit/formatters/sarif.py

```diff
@@ -401,17 +368,13 @@ def create_or_find_rule(issue_dict, rules, rule_indices):
     test_name = issue_dict.get("test_name") or rule_id
     help_uri = docs_utils.get_url(rule_id)
 
-    # Precision based on confidence
     precision = _precision_from_confidence(issue_dict.get("issue_confidence"))
```

Comment:
So this explains your code above. The type-hints above are _wrong_. `confidence` in there is `str | None` in this case. Instead, since this seems to be the usage of that function, it should be `issue_dict.get("issue_confidence", "")` and then you can dramatically simplify the usage of that function. I prefer keeping your current `(str) -> str` signature so let's just change the call site here.